doc/migration/version-upgrade-guide.md

```diff
@@ -168,8 +168,139 @@ The following sections outline the upgrade steps for each SDK version along with
+
+### `dep`
 
+- Update the SDK constraint in `Gopkg.toml` to version `v0.8.2`.
+  ```TOML
+  [[constraint]]
+    name = "github.com/operator-framework/operator-sdk"
+    version = "=v0.8.2"
+  ```
+- Pin the controller-tools dependency to the following revision. See the release notes or [#1278](https://github.com/operator-framework/operator-sdk/pull/1278/) for why this is needed.
+  ```TOML
+  [[override]]
+    name = "sigs.k8s.io/controller-tools"
+    revision = "9d55346c2bde73fb3326ac22eac2e5210a730207"
+  ```
+- Run `dep ensure` to update the vendor directory.
+
+### `modules`
+
+To get familiar with Go modules read the [modules wiki][modules-wiki]. In particular the section on [migrating to modules][migrating-to-modules].
+
+- Ensure that you have Go 1.12+ and [Mercurial][mercurial] 3.9+ installed.
+- Activate Go modules support for your project in `$GOPATH/src` by setting the env `GO111MODULES=on`. See [activating modules][activating-modules] for more details.
+- Initialize a new `go.mod` file by running `go mod init`.
+- Append the following to the end of your `go.mod` file to pin the operator-sdk and other upstream dependencies to the required versions.
+```
+// Pinned to kubernetes-1.13.1
+replace (
+	k8s.io/api => k8s.io/api v0.0.0-20181213150558-05914d821849
+	k8s.io/apiextensions-apiserver => k8s.io/apiextensions-apiserver v0.0.0-20181213153335-0fe22c71c476
+	k8s.io/apimachinery => k8s.io/apimachinery v0.0.0-20181127025237-2b1284ed4c93
+	k8s.io/client-go => k8s.io/client-go v0.0.0-20181213151034-8d9ed539ba31
+)
+
+replace (
+	github.com/coreos/prometheus-operator => github.com/coreos/prometheus-operator v0.29.0
+	k8s.io/code-generator => k8s.io/code-generator v0.0.0-20181117043124-c2090bec4d9b
+	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20180711000925-0cf8f7e6ed1d
+	sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.1.10
+	sigs.k8s.io/controller-tools => sigs.k8s.io/controller-tools v0.1.11-0.20190411181648-9d55346c2bde
+)
+
+replace github.com/operator-framework/operator-sdk => github.com/operator-framework/operator-sdk v0.8.2
+```
+- Run `go mod tidy` to clean up the `go.mod` file. 
+  - In case of any go module loading errors, consult the default [`v0.8.2` go.mod dependencies][v0.8.2-go-mod] scaffolded by the operator-sdk to resolve any differences. You can also view this file by scaffolding a new project with operator-sdk `v0.8.2`.
+- Ensure that you can build the project with `operator-sdk build`
+- Finally remove `Gopkg.lock`, `Gopkg.toml` and the vendor directory.
+
+## `v0.9.x`
+
+- The function `ExposeMetricsPort()` has been replaced with `CreateMetricsService()` [#1560](https://github.com/operator-framework/operator-sdk/pull/1560). 
```

Comment:
I think here we have more changes than that. See [here](https://github.com/operator-framework/operator-sdk-samples/pull/72/commits/f66246e19a1855c34f94c1417559934665110beb#diff-b4731807d0f451420df4f54133ce69f2R7) this file migrate to 0.10 at least. But is required to create a new project with the 0.9.0 and compare. I am not sure if all changes here were introduced in the 0.10 or 0.9.

Reply:
@camilamacedo86 `ExposeMetricsPort()`==>`CreateMetricsService()` is the only breaking change in `v0.8.x` to `v0.9.x`. You can compare the changes here for the scaffolded main.go:
https://github.com/operator-framework/operator-sdk/compare/v0.8.x...v0.9.x#diff-17dd005b8bf50b6dbab49cc2758e0498R162

The PR that you've opened is taking the sample project from `v0.6.0` to `v0.10.0`.
Plus the other changes to the scaffolded cmd/main.go such as [`serveCRMetrics()`](https://github.com/operator-framework/operator-sdk/blob/master/internal/pkg/scaffold/cmd.go#L152-L154) are not breaking changes that are needed to upgrade the versions. It's just an additional feature so I'm not outlining everything that needs to be changed. Just the minimum to avoid breaking changes and dependency updates.

Reply:
Closing as solved 👍